polaris-migrator/src/migrations/replace-sass-focus-ring/replace-sass-focus-ring.ts

```diff
@@ -0,0 +1,191 @@
+// I also don't think we need this
+import {POLARIS_MIGRATOR_COMMENT} from '../../constants';
+// I don't think we need this
+const processed = Symbol('processed');
+
+export type Argument = OrderedArgument | NamedArgument;
+
+export interface NamedArgument {
+  value: string;
+  type: 'named';
+  name: string;
+}
+export interface OrderedArgument {
+  value: string;
+  order: number;
+  type: 'ordered';
+}
+
+type ArgumentBuffer = Node[];
+
+export function getFunctionArgs(node: FunctionNode) {
+  const args: Argument[] = [];
+  const unProcessedArgs: Node[] = node.nodes.slice();
+  let buffer: Node[] | Node[][] = [];
+
+  function resolveArgBuffer(buffer: ArgumentBuffer): Argument {
+    if (!buffer.length) throw Error('Argument buffer must not be empty');
+    const bufferValue = buffer.pop();
+    if (!bufferValue) throw Error('Argument buffer may not contain null value');
+    if (Array.isArray(bufferValue)) {
+      const value = bufferValue.pop();
+      const name = bufferValue.pop();
+      return {
+        type: 'named',
+        value: value.value,
+        name: name.value,
+      } as NamedArgument;
+    } else {
+      return {
+        type: 'ordered',
+        value: bufferValue.value,
+      } as OrderedArgument;
+    }
+  }
+
+  while (unProcessedArgs.length) {
+    const argument = unProcessedArgs.shift() as Node;
+    const isSeparator = argument.type === 'div' && argument.value === ',';
+    const isAssignment = argument.type === 'div' && argument.value === ':';
+    buffer.push(argument as Node);
+    if (isAssignment && unProcessedArgs.length) {
+      // This signifies the middle of a named argument declaration
+      // pop out the next argument as well
+      const key = buffer.shift() as Node;
+      // We empty the buffer here to remove the operator argument
+      // that was previously pushed in
+      buffer = [];
+      const value = unProcessedArgs.shift() as Node;
+      buffer.push([key, value]);
+    }
+
+    if (isSeparator) {
+      // This signifies the end of an argument set
+      // check if the buffer has any arguments in it
+      // if it is an array, convert it into a named argument
+      // and push it into the args array
+      buffer.pop();
+      args.push(resolveArgBuffer(buffer));
+    } else if (!unProcessedArgs.length && buffer.length) {
+      // This is the end of the line,
+      args.push(resolveArgBuffer(buffer));
+    }
+  }
+
+  return args;
+}
+
+interface PluginOptions extends Options, NamespaceOptions {}
+
+const plugin = (options: PluginOptions): Plugin => {
+  const namespacedFocusRing = namespace('focus-ring', options);
+  // TODO FILL IN
+  function declResolver({size = 'base', borderWidth = 0, style = 'base'}) {
+    const offset =
+      borderWidth === 0 ? `rem(1px)` : `(${borderWidth} + rem(1px))`;
+    if (style === 'base') {
+      return [
+        {
+          prop: 'box-shadow',
+          value: `0 0 0 calc(-1 * ${offset}) var(--p-focused)`,
```

Comment:
My only thought is.. should this be a token value 🤔

Reply:
Could we also use the space tokens instead of the `rem` function here? And this could be part of the token.

Should we add `ring` tokens [similar to Tailwind](https://tailwindcss.com/docs/ring-width)? I do like that naming convention, but maybe it isn't semantic enough?

This keeps us true to our token system and removes the worry of having to consider Sass module namespaces for this function.